lib/request.js

```diff
@@ -1,12 +1,24 @@
 'use strict'
 
+// Copied from https://github.com/delvedor/find-my-way/blob/master/index.js#L378
+function sanitizeUrl (url) {
+  for (var i = 0, len = url.length; i < len; i++) {
+    var charCode = url.charCodeAt(i)
+    if (charCode === 63 || charCode === 35) {
+      return url.slice(0, i)
+    }
+  }
+  return url
+}
+
 function Request (params, req, body, query, headers, log) {
   this.params = params
   this.req = req
   this.body = body
   this.query = query
   this.headers = headers
   this.log = log
+  this.path = req ? sanitizeUrl(req.url) : undefined
```

Comment:
when is `req` undefined?

Reply:
Because of [this](https://github.com/fastify/fastify/blob/master/fastify.js#L263) and [this](https://github.com/fastify/fastify/blob/master/lib/request.js#L21)